Cabal/Distribution/Backpack/LinkedComponent.hs

```diff
@@ -185,18 +191,19 @@ toLinkedComponent verbosity db this_pid pkg_map ConfiguredComponent {
 
     -- linked_shape0 is almost complete, but it doesn't contain
     -- the actual modules we export ourselves.  Add them!
-    let reqs = modScopeRequires linked_shape0
-        -- check that there aren't pre-filled requirements...
+    let reqs = Map.keysSet (modScopeRequires linked_shape0)
+                `Set.union` Set.fromList src_reqs
```

Comment:
I guess it is, but for a reason more subtle than reqs including the 'sigantures' field (in the old code, it was mixed in). I'll mention it.